src/it/scala/io/iohk/ethereum/db/IodbDataSourceIntegrationSuite.scala

```diff
@@ -97,19 +81,16 @@ class IodbDataSourceIntegrationSuite extends FunSuite
       keyList.zip(keyListWithExtraByte).foreach { case (key, value) =>
         assert(dbAfterUpdate.get(OtherNamespace, key).contains(value))
       }
+
+      dbAfterUpdate.close()
     }
   }
 
   test("IodbDataSource insert/update with invalid length") {
     forAll(seqByteStringOfNItemsGen(KeySizeWithoutPrefix)) { unFilteredKeyList: Seq[ByteString] =>
-      //create temporary dir
-      val dir = File.createTempFile("iodbInvalidLength", "iodbInvalidLength")
-      dir.delete()
-      dir.mkdir()
-
       val keyList = unFilteredKeyList.take(KeyNumberLimit)
       val (keysLeft, keysToInsert) = Random.shuffle(keyList).splitAt(Gen.choose(0, keyList.size/2).sample.get)
-      val db = new IodbDataSource(new LSMStore(dir = dir, keySize = KeySize))
+      val db = IodbDataSource(path = "/tmp/iodbInvalidLength", keySize = KeySize)
```

Comment:
Could you remove `Random.shuffle` from tests? `keyList` is random anyway. Introuducing randomness to the test makes it impossible to recreate a bug which occurs only for small set of input data.